src/Orckestra.Composer.CustomerConditionProvider/Providers/CustomerConditionProvider.cs

```diff
@@ -0,0 +1,165 @@
+using Orckestra.Tools.ConditionalContent.Types.Config;
+
+namespace Orckestra.Composer.CustomerConditionProvider.Providers
+{
+    public class CustomerConditionProvider : IConditionProvider
+    {
+        protected IComposerContext ComposerContext { get; }
+        protected ICustomerRepository CustomerRepository { get; }
+        protected ICustomerDefinitionsRepository CustomerDefinitionsRepository { get; }
+
+        public CustomerConditionProvider(IComposerContext composerContext,ICustomerRepository customerRepository, ICustomerDefinitionsRepository customerDefinitionsRepository)
+        {
+            ComposerContext = composerContext;
+            CustomerRepository = customerRepository ?? throw new ArgumentNullException(nameof(customerRepository));
+            CustomerDefinitionsRepository = customerDefinitionsRepository ?? throw new ArgumentNullException(nameof(customerDefinitionsRepository));
+        }
+
+        private Dictionary<string, List<string>> Fields => new Dictionary<string, List<string>>()
+        {
+            {
+                "Base", new List<string>()
+                {
+                    nameof(Customer.Language),
+                    nameof(Customer.CustomerType),
+                    nameof(Customer.IsRecurringBuyer)
+                }
+            },
+            {
+                "Activity", new List<string>()
+                {
+                    nameof(Customer.LastLoginDate),
+                    nameof(Customer.LastOrderDate),
+                    nameof(Customer.LastOrderItemsCount),
+                    nameof(Customer.LastOrderLineItemsCount)
+                }
+            }
+        };
+
+        public string Name => "Customer";
+
+        public object GetData()
+        {
+            var customer = GetCustomer();
+
+            return new
+            {
+                Base = customer,
+                Activity = customer,
+                State = new
+                {
+                    IsLoggedIn = ComposerContext.IsAuthenticated 
+                },
+                Custom = customer?.PropertyBag
+            };
+        }
+
+        private Dictionary<string, Field> GetSubFields(EntityDefinition definition, Func<AttributeDefinition, bool> predicate)
+        {
+            return definition.Attributes.Where(predicate)
+                .Select(a => new
+                {
+                    a.Name,
+                    Field = new Field()
+                    {
+                        Label = a.DisplayName?.GetLocalizedValue("en-CA"),
+                        Type = GetQueryFieldType(a)
+                    }
+                })
+                //remove unsupported types
+                .Where(a => a.Field.Type != null)
+                .ToDictionary(a => a.Name, a => a.Field);
+        }
+
+        public Dictionary<string, Field> GetFields()
+        {
+
+            var definition = CustomerDefinitionsRepository.GetCustomerDefinitionAsync().Result;
```

Comment:
Is there caching in place for customer definitions?

Reply:
in repo